src/config/configuration_loader.coffee

```diff
@@ -38,11 +37,16 @@ class ConfigurationLoader
 
 
   loadDefaultConfig: (done) =>
+    fsExtra = require 'fs-extra'
     fsExtra.readJson @defaultConfigPath, done
 
 
   loadUserConfig: (done) =>
-    filePaths = _.map extensions, (ext) => path.join @dir, "dependency-lint.#{ext}"
+    async = require 'async'
```

Comment:
I'm seeing a good amount of repetition here. `async` is defined and imported in many different methods, as is `fs` and `fsExtra`, `path`, etc. 

This approach is similar to the question where variables should be declared (in languages that require that): right before they are used, or all together at the top of the file/method? I personally prefer the former, so in general I like the thinking here. 

Having `require`s inline has the advantage that it becomes much easier to clean out unused dependencies while refactoring code. Dependency-Lint then helps remove them from `package.json`. This is a big advantage. And it makes it easier to see what each variable that is used in code is, and where it comes from. That's big too. 

On the downside, this makes me a bit concerned that different methods use different alternative npm modules. Example: on order to display colored terminal output, method a imports and uses `cli-color`, method 2 uses `colors`, and method 3 uses `chalk`. Because they are not aware that the other method already uses a library. Arguably, if we keep file sizes at a sane level, this argument isn't very strong.